elements/ethernet/checkarpheader.cc

```diff
@@ -60,24 +64,28 @@ CheckARPHeader::configure(Vector<String> &conf, ErrorHandler *errh)
         .read("VERBOSE", verbose)
         .read("DETAILS", details)
         .complete() < 0)
-    return -1;
+        return -1;
 
     _verbose = verbose;
-    if (details)
-    _reason_drops = new atomic_uint32_t[NREASONS];
+    if (details) {
+        _reason_drops = new atomic_uint64_t[NREASONS];
+        memset(_reason_drops, 0, NREASONS * sizeof(atomic_uint64_t));
+    }
 
     return 0;
 }
 
 Packet *
 CheckARPHeader::drop(Reason reason, Packet *p)
 {
-    if (_drops == 0 || _verbose)
+    if (_drops == 0 || _verbose) {
         click_chatter("ARP header check failed: %s", reason_texts[reason]);
+    }
     _drops++;
 
-    if (_reason_drops)
+    if (_reason_drops) {
         _reason_drops[reason]++;
+    }
 
     checked_output_push(1, p);
```

Comment:
This one is problematic. Under batch mode it will send a single packet as non-batch and lead to a crash.

Reply:
But then it's not your fault, it was already there :p Can you just fix it?

Reply:
Yes, I will fix it